state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -40,12 +40,14 @@ const BASIS_POINTS_PER_MILLION: u32 = 100;
 pub const SWAP_DELAY_BLOCKS: u32 = 2;
 
 #[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
 pub enum SwapType {
 	Swap(ForeignChainAddress),
 	CcmPrincipal(SwapId),
 	CcmGas(SwapId),
 }
 #[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Sorry forgot to mention this. This serde derive is not used, and so can be removed.